src/main/java/org/wildfly/security/auth/util/GSSCredentialSecurityFactory.java

```diff
@@ -322,9 +323,16 @@ private GSSKerberosCredential createGSSCredential(Configuration configuration) t
             final Subject subject = new Subject();
 
             try {
-                final LoginContext lc = new LoginContext("KDC", subject, (c) -> {
-                    throw new FastUnsupportedCallbackException(c[0]);
-                }, configuration);
+                final ClassLoader oldCl = WildFlySecurityManager.getCurrentContextClassLoaderPrivileged();
+                WildFlySecurityManager.setCurrentContextClassLoaderPrivileged(Builder.class.getClassLoader());
```

Comment:
Does that override jdk behaviour. I mean OpenJDK use explicitelly Thread.currentThread().getContextClassLoader() [1] 
[1] http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/javax/security/auth/login/LoginContext.java#278

Reply:
Yes these methods are shortcuts to avoid the necessity of a privileged block.

Reply:
I see